engine/errors.go

```diff
@@ -0,0 +1,51 @@
+package engine
+
+import "errors"
+
+var (
+	// ErrFeatureNotSupported is a common error that may be returned from optional
+	// Engine methods to indicate the engine implementation doesn't support the
+	// given feature.
+	//
+	// Note, all methods are allowed to return this error, some methods are
+	// required, and may not return this error.
+	//
+	// When the worker encounters this error from an optional method, it should
+	// workaround if possible, but most likely resolve the task as "exception"
+	// with reason "malformed-payload".
+	ErrFeatureNotSupported   = errors.New("Feature not support by current engine")
+	ErrResourceNotFound      = errors.New("The referenced resource wasn't found")
+	ErrSandboxTerminated     = errors.new("The Sandbox has terminated")
+	ErrSandboxAborted        = errors.new("Exection of sandbox was aborted")
+	ErrNonFatalInternalError = errors.new("Engine encountered a non-fatal internal error")
+	ErrContractViolation     = errors.new("Engine has detected a contract violation")
+	ErrEngineIsSingleton     = errors.New("Engine cannot run multiple sandboxes in parallel")
+	ErrEngineNotSupported    = errors.New("Engine is not available in the current configuration")
+	ErrEngineUnknown         = errors.New("Engine with the given doesn't exist")
+)
```

Comment:
Errors will need instance data with them for them to be helpful. For example, an error that says `Feature not supported by current engine` is much harder to troubleshoot than an error `Feature "livelog" not supported by engine "OSX-jail-worker-v1"`. Therefore I think we can create different go types for each class of error, but the errors themselves should keep references to the instance data, so when the `.String()` method is called to generate the error output, we see all the relevant information. This way, the _type_ would map to one of these classes of error, but the _instance_ would contain references to the actual error that occurred.

Let's therefore define types here, and not instantiate any variables.